test/e2e/scripts/generate-parameters.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+##### A script to generate a unique namespace #####
+##### and a parameters file for a workflow    #####
+
+
+##### Exit on error #####
+set -e
+
+##### Functions #####
+
+usage()
+{
+    echo '    Usage: ./generate_parameters.sh [[-w workflow -g workflow_group] | [-h]]
+    Example: ./generate_parameters.sh -g workflow_group -w workflow
+    Flags:
+    -w, --workflow         workflow name
+    -g, --workflow-group   workflow group name
+    -o, --output-file      generated yaml file name (default parameters.yaml)
+    -d, --workflows-dir    the directory where workflows are defined (default ../argo-workflows)'
+}
+
+check_yq_installed()
+{
+    if ! [ -x "$(command -v yq)" ]; then
+        echo 'Error: yq is not installed.'
+        exit 1
+    fi
+}
+
+validate_input()
+{
+    # Validate workflow name characters
+    if ! [[ $WORKFLOW =~ ^[0-9a-zA-Z._-]+$ ]]; then
+        echo 'Error: Invalid workflow name format: '$WORKFLOW
+        exit 1
+    fi
+
+    # Validate workflow group name characters
+    if ! [[ $WORKFLOW_GROUP =~ ^[0-9a-zA-Z._-]+$ ]]; then
```

Comment:
Same thing here.

Reply:
I'd allow the workflow group directory name to contain `_` or `.` then serialize it when generating the namespace name https://github.com/DataDog/datadog-agent/pull/3710/files#diff-588d687f46319d145cc2508f71d80f80R53

Reply:
SGTM